internal/reader/rmn_home_test.go

```diff
@@ -53,80 +54,97 @@ func TestRMNHomeChainConfigPoller_Ready(t *testing.T) {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			homeChainReader := readermock.NewMockContractReaderFacade(t)
 
-	// Set up the mock for a successful call after 10 errors
-	primaryConfig, secondaryConfig := createTestRMNHomeConfigs(false, false)
-	rmnHomeOnChainConfigs := []VersionedConfigWithDigest{primaryConfig, secondaryConfig}
+			// Create a variable to track if GetLatestValue was called
+			var getLatestValueCalled bool
+			var mu sync.Mutex
 
-	homeChainReader.On(
-		"GetLatestValue",
-		mock.Anything,
-		mock.Anything,
-		mock.Anything,
-		mock.Anything,
-		mock.Anything,
-	).Run(func(args mock.Arguments) {
-		arg := args.Get(4).(*[]VersionedConfigWithDigest)
-		*arg = rmnHomeOnChainConfigs
-	}).Return(nil).Times(50)
+			homeChainReader.On("GetLatestValue",
+				mock.Anything,
+				mock.Anything,
+				mock.Anything,
+				mock.Anything,
+				mock.Anything,
+			).Run(func(args mock.Arguments) {
+				result := args.Get(4).(*[]VersionedConfigWithDigest)
+				*result = []VersionedConfigWithDigest{
+					{
+						ConfigDigest: [32]byte{1},
+						VersionedConfig: VersionedConfig{
+							Version: 1,
+							Config: Config{
+								Nodes:        []Node{},
+								SourceChains: []SourceChain{},
+							},
+						},
+					},
+				}
+				mu.Lock()
+				getLatestValueCalled = true
+				mu.Unlock()
+			}).Return(nil)
 
-	homeChainReader.On(
-		"GetLatestValue",
-		mock.Anything,
-		mock.Anything,
-		mock.Anything,
-		mock.Anything,
-		mock.Anything,
-	).Return(fmt.Errorf("error"))
+			poller := NewRMNHomePoller(
+				homeChainReader,
+				rmnHomeBoundContract,
+				logger.Test(t),
+				10*time.Millisecond,
+			).(*rmnHomePoller)
 
-	configPoller := NewRMNHomePoller(
-		homeChainReader,
-		rmnHomeBoundContract,
-		logger.Test(t),
-		tickTime,
-	)
+			require.NoError(t, poller.Start(context.Background()))
 
-	require.NoError(t, configPoller.Start(context.Background()))
-	// Initially it's healthy
-	healthy := configPoller.HealthReport()
-	require.Equal(t, map[string]error{configPoller.Name(): error(nil)}, healthy)
-
-	// give some time for polling to happen
-	time.Sleep(totalSleepTime)
-	errors := configPoller.HealthReport()
-	require.Equal(t, 1, len(errors))
-	require.Errorf(t, errors[configPoller.Name()], "polling failed %d times in a row", MaxFailedPolls)
-
-	// give some time for successful polling to happen
-	time.Sleep(totalSleepTime * 1)
-	errors = configPoller.HealthReport()
-	require.Equal(t, 1, len(errors))
-	require.Equal(t, map[string]error{configPoller.Name(): error(nil)}, healthy) // should not produce an error
-
-	// give some time for polling to fail again
-	time.Sleep(totalSleepTime * 1)
-	errors = configPoller.HealthReport()
-	require.Equal(t, 1, len(errors))
-	require.Errorf(t, errors[configPoller.Name()], "polling failed %d times in a row", MaxFailedPolls)
+			// Wait for the initial fetch to complete
+			require.Eventually(t, func() bool {
+				mu.Lock()
+				defer mu.Unlock()
+				return getLatestValueCalled
```

Comment:
Ah I see, ok this is interesting.

Reply:
I think you can check if something is `Called` using the mock? Lemme check the API.

Reply:
This is working for me, inside the `func() bool` for the `Eventually`:

```golang
return homeChainReader.AssertCalled(t, "GetLatestValue", mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything)
```

Reply:
Nice one! thanks